examples/linux/io_epoll_test.cpp

```diff
@@ -84,6 +114,105 @@ int main() {
   } catch (const std::exception& ex) {
     std::printf("error: %s\n", ex.what());
   }
+
+  auto pipe = open_pipe(scheduler);
+
+  inplace_stop_source stopWarmup;
+  inplace_stop_source stopRead;
+  inplace_stop_source stopWrite;
+  auto buffer = std::vector<char>{};
+  buffer.resize(1);
+  auto offset = 0;
+  auto reps = 0;
+  const auto databuffer = as_bytes(span{data});
+  std::thread w{[&, &wPipe = std::get<1>(pipe)] () { 
+      try {
+        printf("writes starting!\n");
+        while(!stopWrite.stop_requested()) {
+          sync_wait(
+            async_write_some(wPipe, databuffer), 
+            stopWrite.get_token());
+        }
+        printf("writes exiting!\n");
+      } catch (const std::error_code& ec) {
+        std::printf("async_write_some error: %s\n", ec.message().c_str());
+      } catch (const std::exception& ex) {
+        std::printf("async_write_some exception: %s\n", ex.what());
+      }
+    }};
+  scope_guard waitForWrites = [&]() noexcept {
+    stopWrite.request_stop();
+    printf("writes stopping..!\n");
+    w.join();
+    printf("writes stopped!\n");
+  };
+  auto pipe_bench = [&, &rPipe = std::get<0>(pipe)](int seconds, auto& stopSource) {
+    return unstoppable(
+      with_query_value(
+        discard(
+          when_all(
+            transform(
+              defer(
+                [&, seconds](){
+                  return schedule_at(scheduler, now(scheduler) + std::chrono::seconds(seconds));
+                }),
+              [&]{
+                  stopSource.request_stop();
+              }),
+            typed_via(
+              repeat(
+                transform(
+                  discard(
+                    async_read_some(rPipe, as_writable_bytes(span{buffer.data() + 0, 1}))),
+                  [&]{
+                    assert(data[(reps + offset)%sizeof(data)] == buffer[0]);
+                    ++reps;
+                  })), 
+              scheduler)
+          )),
+          get_stop_token, stopSource.get_token()));
+  };
+  auto start = std::chrono::high_resolution_clock::now();
+  auto end = std::chrono::high_resolution_clock::now();
+  try {
+    sync_wait(
+      sequence(
+        pipe_bench(WARMUP_DURATION, stopWarmup), // warmup
+        lazy([&]{
+          // restart reps and keep offset in data
+          offset = reps%sizeof(data);
+          reps = 0;
+          printf("warmup completed!\n");
+          // exclude the warmup time
+          start = std::chrono::high_resolution_clock::now();
```

Comment:
I would rather have a more general timestamp + materialize + subject + tap, so that inserting `time("named", sub)` would allow `sync_wait(reduce(filter(sub, [](auto cpo, string id, auto at){return id == "name";}), startTofinishDuration))`.